risc0/zkvm/src/host/server/exec/syscall/pipe.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright 2024 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{
+    cell::RefCell,
+    collections::VecDeque,
+    io::{Read, Write},
+    rc::Rc,
+};
+
+use anyhow::{anyhow, Result};
+
+use crate::host::client::posix_io::PosixIo;
+
+use super::{Syscall, SyscallContext};
+
+// This is an arbitrary maximum number of open file descriptors allowed.
+const MAX_FD: u32 = 1000;
```

Comment:
Would we want to make this configurable in the future?

Reply:
Perhaps